js/filter.js

```diff
@@ -0,0 +1,86 @@
+/** @format */
+
+const refsFilter = {
+  form: document.querySelector(".search_form"),
+  search: document.querySelector(".search_imput"),
+  radioAuthor: document.getElementById("author"),
+  radioMovie: document.getElementById("movie"),
+  content: document.querySelector(".content"),
+};
+let query = "";
+
+const noResult = createElem({
+  nodeType: "p",
+  className: "no_result",
+  text: "NO RESULT",
+});
+
+const searchData = (value, query, path) => {
+  refsFilter.content.innerHTML = "";
+
+  fetchData(
+    `search/${value}?api_key=${API_KEY}&query=${query}&language=en-US&page=1&include_adult=false`
+  ).then((result) => {
+    if (!result.total_results) {
+      refsFilter.content.innerHTML = "";
+      refsFilter.content.insertAdjacentElement("beforeend", noResult);
+      return;
+    }
+    path ? createBlog(result.results[0].known_for) : createBlog(result.results);
+  });
+};
+
+const getInputValue = () => {
+  try {
+    const savedStrQuery = localStorage.getItem("query");
+    const savedStrRadio = localStorage.getItem("checked");
+    if (savedStrQuery) {
+      refsFilter.search.value = savedStrQuery;
+    }
+    if (savedStrRadio === "author") {
+      refsFilter.radioAuthor.checked = "checked";
+      searchData("person", savedStrQuery, 1);
+    }
+    if (savedStrRadio === "movie") {
+      refsFilter.radioMovie.checked = "checked";
+      searchData("movie", savedStrQuery, 0);
+    }
+  } catch (error) {
+    console.log(error.message);
+    return;
+  }
+};
+
+getInputValue();
+
+const changeNamePlaceholder = (event) => {
+  const { type, dataset, id } = event.target;
+  let checkedImput = "";
+  if (type === "radio") {
+    refsFilter.search.placeholder = `Search by ${dataset.placeholder}`;
+    checkedImput = id;
+  }
+  localStorage.setItem("checked", checkedImput);
+};
+
+refsFilter.form.addEventListener("change", (event) => {
```

Comment:
You can just pass the changeNamePlaceholder function, like this:
```suggestion
refsFilter.form.addEventListener("change", changeNamePlaceholder);
```